code/graphics/debug_sphere.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include "globalincs/pstypes.h"
+struct debug_sphere {
+	vec3d pos;
+	float radius;
+};
+
+namespace debug_spheres {
```

Comment:
Did you think about making this a class?  You're using a very modular design here, which is excellent, but it could be even further improved, imo, by making this a class with the sphere vector being a member and the global variable being of the class instead.  The advantage would be that you would automatically protect the sphere vector from accidental outside edits, and you could mark `clear()` as private, while keeping `add()` and `render()` as public functions as well.  Again, this avoids some other chunk of code clearing the vector because a bug got through.

Reply:
I did think about putting it all into a class and doing things like that. I'm not sure why I went with a namespace, I'd probably just been reminded they exist by another part of the code and was trying to mimic conventions. When in Rome and so forth. I'll happily switch it over, in any case.

Reply:
You can even do both, but moving it over is the best way to keep anyone from accidentally adding an inappropriate `clear()` call.

Reply:
I've reorganized everything into a class with Spheres a private static field. If there's any further adjustment it could use just let me know. My lighting data branch uses a similar pattern, so it'll be to good to know if this is solid or needs tweaking.